### PR TITLE
Mesh crd from discovery

### PR DESCRIPTION
**Pull request by EItanya**

### CODE REVIEW

File: pkg/install/common/mesh_install_syncer.go [hunk 1]
```diff
@@ -133,3 +148,30 @@ func (s *MeshInstallSyncer) handleActiveInstalls(ctx context.Context,
...
+							Namespace: mesh.Metadata.Namespace,
+						}
+					}
+				case *v1.Mesh_LinkerdMesh:
+					if meshType.LinkerdMesh.InstallationNamespace == in.InstallationNamespace &&
+						mesh.Metadata.Name == in.Metadata.Name {
+						meshInstall.InstalledMesh = &core.ResourceRef{
+							Name:      mesh.Metadata.Name,
+							Namespace: mesh.Metadata.Namespace,
+						}
```
Comment (ilackarms) [inline pkg/install/common/mesh_install_syncer.go]: reduce the code duplication here: 
make the switch statement retrieve the `installNampespace` of the install, then do the loop over the meshes in one code block

File: pkg/install/common/mesh_install_syncer.go [hunk 2]
```diff
@@ -133,3 +148,30 @@ func (s *MeshInstallSyncer) handleActiveInstalls(ctx context.Context,
 	}
 	return nil, nil
 }
+
+func addMeshToInstall(in *v1.Install, meshes v1.MeshList) {
```
Comment (ilackarms) [inline pkg/install/common/mesh_install_syncer.go]: in a follow up pr, we should remove this func as well as the `installedMesh` field from install crd

File: pkg/install/istio/istio_installer.go [hunk 3]
```diff
@@ -76,37 +75,7 @@ func (i *defaultIstioInstaller) EnsureIstioInstall(ctx context.Context, install
...
-				InstallationNamespace: installNamespace,
-				IstioVersion:          istio.IstioVersion,
-			},
-		},
-		MtlsConfig: &v1.MtlsConfig{
-			MtlsEnabled:     istio.EnableMtls,
-			RootCertificate: istio.CustomRootCert,
-		},
-	}
+	return nil, nil
```
Comment (ilackarms) [inline pkg/install/istio/istio_installer.go]: if this func always `nil`, why do we have `*v1.Mesh` as a return type?

Comment (EItanya) [reply pkg/install/istio/istio_installer.go]: I tried to remove it, but it won't be possible until linkerd is discovered as well

File: pkg/meshdiscovery/pkg/discovery/istio/istio.go [hunk 4]
```diff
@@ -38,9 +38,24 @@ func filterIstioMeshes(meshes v1.MeshList) v1.MeshList {
 	return result
 }
 
+func filterIstioInstalls(installs v1.InstallList) v1.InstallList {
```
Comment (ilackarms) [inline pkg/meshdiscovery/pkg/discovery/istio/istio.go]: nit: rename to `getIstioInstalls` (filter implies removing)

File: pkg/meshdiscovery/pkg/discovery/istio/istio.go [hunk 5]
```diff
@@ -104,9 +119,11 @@ func mergeMeshes(discoveredMeshes, existingMeshes v1.MeshList) (v1.MeshList, err
 			},
 		}
 		discoveredMesh.MtlsConfig = discoveredMesh.DiscoveryMetadata.MtlsConfig
-		discoveredMesh.Metadata = core.Metadata{
-			Namespace: getWriteNamespace(),
-			Name:      fmt.Sprintf("istio-%s", discoveredMesh.DiscoveryMetadata.InstallationNamespace),
+		if discoveredMesh.Metadata.Name == "" || discoveredMesh.Metadata.Namespace == "" {
+			discoveredMesh.Metadata = core.Metadata{
+				Namespace: getWriteNamespace(),
+				Name:      fmt.Sprintf("istio-%s", discoveredMesh.DiscoveryMetadata.InstallationNamespace),
```
Comment (ilackarms) [inline pkg/meshdiscovery/pkg/discovery/istio/istio.go]: note that this naming convention will need to be updated when we move to multi cluster

File: pkg/meshdiscovery/pkg/discovery/istio/istio.go [hunk 6]
```diff
@@ -131,6 +148,31 @@ func constructDiscoveryData(ctx context.Context, istioPilotPod *v1.Pod) (*v1.Mes
 		InjectedNamespaceLabel: injectionLabel,
 	}
 	mesh.DiscoveryMetadata = discoveryData
+
+	// If install crd exists, overwrite discovery data
+	for _, install := range existingInstalls {
```
Comment (ilackarms) [inline pkg/meshdiscovery/pkg/discovery/istio/istio.go]: this loop should probably be made into a utility function that we can use in the follow up PR to remove `installedMesh`

Comment (EItanya) [reply pkg/meshdiscovery/pkg/discovery/istio/istio.go]: As a part of my new PR I made all of this code into utilities

File: pkg/meshdiscovery/pkg/discovery/istio/istio.go (hunk 5 above)
Comment (EItanya) [reply pkg/meshdiscovery/pkg/discovery/istio/istio.go]: Agreed, a lot of changes will probably have to be made

File: pkg/install/istio/istio_installer.go (hunk 3 above)
Comment (ilackarms) [reply pkg/install/istio/istio_installer.go]: the interface is `EnsureIstioInstall(ctx context.Context, install *v1.Install, meshes v1.MeshList) (*v1.Mesh, error)`

Comment (ilackarms) [reply pkg/install/istio/istio_installer.go]: not shared

**Pull request merged**